### PR TITLE
Improve documentation for installation / quick start

### PR DESCRIPTION
**Pull request by nelimee**
This PR moves the installation instructions to the sphinx-generated documentation website. It also adds a "Quick start" page with a step-by-step guide on how to build the circuit for a CNOT gate. 

For the moment, the step-by-step guide is using #321, but depending on the choices we make in the future it might have to change to use #298. 

It also follows https://daiyi.co/blog/pr-previews-for-github-pages/ to be able to preview the documentation on a PR without impacting (too much) the main documentation.

### CODE REVIEW

**Comment by nelimee** [comment]
This PR might need very minor changes once #321 will be merged (maybe change of some URLs or some code examples), but should be quite ready to review.

**Comment by KabirDubey** [comment]
I can work through these instructions and check for bugs sometime this weekend!

**Comment by nelimee** [comment]
> I can work through these instructions and check for bugs sometime this weekend!

A few notes:
- I made that live in the last meeting.
- you'll have to change the instructions slightly to clone the branch from #321 (see the recording of this week meeting).

**Comment by KabirDubey** [comment]
> A few notes:
> 
> * I made that live in the last meeting.
> * you'll have to change the instructions slightly to clone the branch from [Compile `BlockGraph` to `stim.Circuit` #321](https://github.com/QCHackers/tqec/pull/321) (see the recording of this week meeting).

Thanks! I was present during your demo; my thinking is I review this PR by seeing if your instructions work on my machine (OSX).

File: docs/quick_start.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+4. Compile and export the computation
+-------------------------------------
+
+In order to get a ``stim.Circuit`` instance, the computation first need to be compiled.
+
+.. code-block:: python
+
+    from tqec import compile_block_graph
+
+    compiled_computation = compile_block_graph(block_graph, [observables[1]])
```
Comment (KabirDubey) [inline docs/quick_start.rst:67]: An explanation for the `observables` field here would be useful. I'm not sure if how the choice of the second out of 3 observables matters.

**Review by KabirDubey — APPROVED** [review]
Left some comments but overall LGTM! I'm unfamiliar with the GitHub pages stuff so, if it's important, perhaps someone else can take a look before we merge. 

I successfully reproduced the results from the 9/25 demo. 

I tried to add some more code at the end of the `main.py` file to reproduce the logical CNOT error rate [graph](https://github.com/QCHackers/tqec/pull/321#issuecomment-2371349466) from #321 and save it to the same `images` directory as the observable pictures. IIRC this graph is ultimately our desired output so I think we should include it in our usage demo. I wasn't able to get modify the code from #321 without error but I can assign myself to the task if you agree.

**Comment by nelimee** [comment]
> Left some comments but overall LGTM! I'm unfamiliar with the GitHub pages stuff so, if it's important, perhaps someone else can take a look before we merge. 
> 
> I successfully reproduced the results from the 9/25 demo. 
> 
> I tried to add some more code at the end of the `main.py` file to reproduce the logical CNOT error rate [graph](https://github.com/QCHackers/tqec/pull/321#issuecomment-2371349466) from #321 and save it to the same `images` directory as the observable pictures. IIRC this graph is ultimately our desired output so I think we should include it in our usage demo. I wasn't able to get modify the code from #321 without error but I can assign myself to the task if you agree.

Nice thanks!
I was able to reproduce the results from #321. I will add your suggestions and merge that PR tomorrow.

**Pull request merged**